integration/flink/flink2/src/test/java/io/openlineage/flink/visitor/identifier/catalog/GenericInMemoryCatalogSymlinkProviderTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.flink.visitor.identifier.catalog;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.openlineage.client.utils.DatasetIdentifier.SymlinkType;
+import org.apache.flink.streaming.api.lineage.LineageDataset;
+import org.apache.flink.table.catalog.AbstractCatalog;
+import org.apache.flink.table.catalog.GenericInMemoryCatalog;
+import org.apache.flink.table.catalog.listener.CatalogContext;
+import org.junit.jupiter.api.Test;
+
+public class GenericInMemoryCatalogSymlinkProviderTest {
+
+  GenericInMemoryCatalogSymlinkProvider provider = new GenericInMemoryCatalogSymlinkProvider();
+  CatalogContext catalogContext = mock(CatalogContext.class);
+  LineageDataset lineageDataset = mock(LineageDataset.class);
+
+  @Test
+  void testIsDefined() {
+    assertThat(provider.isDefinedAt(GenericInMemoryCatalog.class)).isTrue();
+    assertThat(provider.isDefinedAt(AbstractCatalog.class)).isFalse();
+  }
+
+  @Test
+  void testGetSymlink() {
+    when(lineageDataset.name()).thenReturn("catalogName.tableName");
+    when(catalogContext.getCatalogName()).thenReturn("catalogName");
+    assertThat(provider.getSymlink(catalogContext, lineageDataset))
+        .isPresent()
+        .get()
+        .hasFieldOrPropertyWithValue("namespace", "flink://")
+        .hasFieldOrPropertyWithValue("name", "tableName")
+        .hasFieldOrPropertyWithValue("type", SymlinkType.TABLE);
+  }
+
+  @Test
+  void testGetSymlinkRemovesOnlyCatalogFromDatasetName() {
+    when(lineageDataset.name()).thenReturn("catalogName.catalogName.tableName");
+    when(catalogContext.getCatalogName()).thenReturn("catalogName");
+    assertThat(provider.getSymlink(catalogContext, lineageDataset))
+        .isPresent()
+        .get()
+        .hasFieldOrPropertyWithValue("namespace", "flink://")
```

Comment:
Doesn't flink have an address?

Reply:
Perhaps it's available, but still this something like the address of Spark driver. There is not much value out of it.

Reply:
In this case Flink instances running on different hosts will produce the same symlink, although they can be independent, and lineage graph could become misleading

Reply:
I still think it's the same as with local Spark pointing to `file`. Globally this can be misleading, but it is convenient for testing.

Reply:
The other solution would be to remove the symlink in this scenario for `GenericInMemoryCatalog` while leaving the `CatalogSymlinkProvider` for future extensions.

Reply:
Could we discuss this in #3642?